src/components/collection/DataPreview/index.tsx

```diff
@@ -0,0 +1,107 @@
+import RefreshIcon from '@mui/icons-material/Refresh';
+import {
+    Alert,
+    AlertTitle,
+    Button,
+    LinearProgress,
+    Stack,
+    ToggleButton,
+    ToggleButtonGroup,
+    Typography,
+} from '@mui/material';
+import ListView from 'components/collection/DataPreview/ListView';
+import TableView from 'components/collection/DataPreview/TableView';
+import { useJournalData, useJournalsForCollection } from 'hooks/useJournalData';
+import { useLiveSpecs_spec } from 'hooks/useLiveSpecs';
+import { useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { hasLength } from 'utils/misc-utils';
+
+interface Props {
+    collectionName: string;
+}
+
+enum Views {
+    table = 'table',
+    list = 'list',
+}
+
+export function DataPreview({ collectionName }: Props) {
+    const [previewMode, setPreviewMode] = useState<Views>(Views.list);
+    const toggleMode = (_event: any, newValue: Views) => {
+        setPreviewMode(newValue);
+    };
+
+    const { liveSpecs: publicationSpecs } = useLiveSpecs_spec(
+        `datapreview-${collectionName}`,
+        [collectionName]
+    );
+    const spec = useMemo(() => publicationSpecs[0], [publicationSpecs]);
+
+    // TODO (typing) we need to fix typing
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    const journals = useJournalsForCollection(spec?.catalog_name);
+    const { data: journalsData, isValidating: journalsLoading } = journals;
+
+    // TODO (typing) we need to fix typing
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    const journal = useMemo(() => journalsData?.journals?.[0], [journalsData]);
```

Comment:
If the collection is partitioned, this will only select a single partition. This is absolutely a reasonable compromise for now. IMO this type of thing is worth a comment, though, since otherwise it just looks "wrong".

Reply:
I'd love to be able to read data from all partitions. That's part of the flowctl-in-Rust work you're doing, right @psFried?

Reply:
I'm doing essentially the same thing that you did here for `flowctl`, at least for the short term :grin: But yea, the intention is definitely to make that part of the Rust client.